src/functionalTest/java/uk/gov/hmcts/sptribs/testutil/IdamTokenGenerator.java

```diff
@@ -38,24 +42,51 @@ public class IdamTokenGenerator {
     @Autowired
     private IdamClient idamClient;
 
+    private final Cache<String, String> cache = Caffeine.newBuilder().expireAfterWrite(2, TimeUnit.HOURS).build();
+
     public String generateIdamTokenForSolicitor() {
-        return idamClient.getAccessToken(solicitorUsername, solicitorPassword);
+        String solicitorUserToken = cache.getIfPresent(solicitorUsername);
+        if (solicitorUserToken == null) {
+            solicitorUserToken = idamClient.getAccessToken(solicitorUsername, solicitorPassword);
+            cache.put(solicitorUsername, solicitorUserToken);
+        }
+        return solicitorUserToken;
     }
 
     public String generateIdamTokenForCaseworker() {
-        return idamClient.getAccessToken(caseworkerUsername, caseworkerPassword);
+        String caseworkerUserToken = cache.getIfPresent(caseworkerUsername);
+        if (caseworkerUserToken == null) {
+            caseworkerUserToken = idamClient.getAccessToken(caseworkerUsername, caseworkerPassword);
+            cache.put(caseworkerUsername, caseworkerUserToken);
+        }
+        return caseworkerUserToken;
     }
 
-    public String generateIdamTokenForSystem() {
-        return idamClient.getAccessToken(systemUpdateUsername, systemUpdatePassword);
+    public String generateIdamTokenForSystemUser() {
```

Comment:
Is this renamed method not called anywhere? Would instances of calling the original method not need to be renamed too ('SystemUser' instead of 'System')?

Also, how is the cache going to be tested? Or are there statistics available?